packages/hooks/src/data/QueryData.ts

```diff
@@ -206,8 +207,8 @@ export class QueryData<TData, TVariables> extends OperationData {
     // See if there is an existing observable that was used to fetch the same
     // data and if so, use it instead since it will contain the proper queryId
     // to fetch the result set. This is used during SSR.
-    if (this.context && this.context.renderPromises) {
-      this.currentObservable.query = this.context.renderPromises.getSSRObservable(
+    if (this.ssrInitiated()) {
+      this.currentObservable.query = this.context?.renderPromises?.getSSRObservable(
```

Comment:
The optional chaining operator is a nice addition here.

Reply:
Thanks @jasonwr - I agree. I've started sprinkling its use in, but there are definitely more places it can be used for sure.